server/s3_put_multiobject_copy_action.cc

```diff
@@ -360,10 +345,79 @@ void S3PutMultipartCopyAction::initiate_part_copy() {
   if (additional_object_metadata->get_number_of_fragments() == 0) {
     copy_part_object();
   } else {
-    s3_copy_part_action_state = S3PutObjectActionState::validationFailed;
-    set_s3_error("NotImplemented");
+    copy_multipart_source_object();
+  }
+}
+
+void S3PutMultipartCopyAction::copy_multipart_source_object() {
+  s3_log(S3_LOG_INFO, stripped_request_id, "%s Entry\n", __func__);
+  if (!total_data_to_copy) {
+    s3_log(S3_LOG_DEBUG, stripped_request_id, "Source object is empty");
+    next();
+    return;
+  }
+  object_data_copier.reset(new S3ObjectDataCopier(
+      request, motr_writer, motr_reader_factory, s3_motr_api));
+
+  extended_obj = additional_object_metadata->get_extended_object_metadata();
+  if (additional_object_metadata->get_number_of_parts() == 0) {
+    // FNO + base object itself as fragment
+    total_objects = additional_object_metadata->get_number_of_fragments() + 1;
```

Comment:
Going by code below, this 'total_objects' means 'total_parts' i guess, because this nomenclature is creating confusion

Reply:
Cleaned up code. This is removed